controllers/secretproviderclasspodstatus_controller.go

```diff
@@ -19,49 +19,136 @@ package controllers
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
 	"strings"
+	"sync"
 	"time"
 
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-	"k8s.io/apimachinery/pkg/util/wait"
+	log "github.com/sirupsen/logrus"
 
+	"sigs.k8s.io/secrets-store-csi-driver/apis/v1alpha1"
+	"sigs.k8s.io/secrets-store-csi-driver/pkg/util/fileutil"
+	"sigs.k8s.io/secrets-store-csi-driver/pkg/util/secretutil"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
-	log "github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/secrets-store-csi-driver/apis/v1alpha1"
-)
-
-const (
-	certType       = "CERTIFICATE"
-	privateKeyType = "RSA PRIVATE KEY"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 // SecretProviderClassPodStatusReconciler reconciles a SecretProviderClassPodStatus object
 type SecretProviderClassPodStatusReconciler struct {
 	client.Client
+	Mutex  *sync.Mutex
 	Log    *log.Logger
 	Scheme *runtime.Scheme
 	NodeID string
 	Reader client.Reader
 	Writer client.Writer
 }
 
+func (r *SecretProviderClassPodStatusReconciler) RunPatcher(stopCh <-chan struct{}) {
```

Comment:
The `Patcher` doesn't need to part of this PR. I've included it as an optimization based on load tests. If required, I can stub this change out from this PR and create a new PR to just add the `Patcher` to keep reviews easy.

Reply:
Are we patching here periodically in case the ownerReference gets removed from the synced secrets?

Reply:
Yeah, this patcher adds the `ownerReferences` to the secret in batches. Periodically checks to see if any of the required `ownerRefs` are missing and adds them.